chore(banner): update a11y docs (#3260)
patternfly/patternfly-org: packages/v4/patternfly-docs/content/accessibility/banner/banner.md

section: components
---

import { Checkbox, List, ListItem } from '@patternfly/react-core';

## Accessibility

To implement an accessible PatternFly **banner**:

- Check the color contrast of a banner and any text on it to make sure they can be read by users with color blindness or other visual impairments.

> - When a banner is used to convey status, add an icon that also conveys status visually, and provide screen reader text using `.pf-screen-reader` or an equivalent.

Reviewer: is it `.pf-screen-reader`? or `.pf-u-screen-reader`?

## Testing

At a minimum, an banner should meet the following criteria:

<List isPlain>
  <ListItem>
    <Checkbox id="banner-a11y-checkbox-1" label="A user should not be able to focus on the banner with keyboard controls." />
  </ListItem>
  <ListItem>
    <Checkbox id="banner-a11y-checkbox-2" label="A user should be able to have a screen reader describe the contents of the banner, but not focus on or interact with the banner otherwise." />
  </ListItem>
  <ListItem>

>     <Checkbox id="banner-a11y-checkbox-3" label={<span>If the banner conveys status, a status icon and <code className="ws-code">.pf-screen-reader</code> element is included.</span>} />

Reviewer 2: Bummer we can't use markdown in react props like this. I noticed this format in the accordion docs, too. I'm guessing there isn't a way around it? Passing the prop values as children is the only way I have seen how to do it. Just seems like it will lead to inconsistencies.

https://github.com/patternfly/patternfly-org/blob/42a6910f36c412a428852947599154935f0ff1ef/packages/v4/patternfly-docs/content/accessibility/accordion/accordion.md#L26-L42

  </ListItem>
</List>

## React customization

The following React props have been provided for more fine-tuned control over accessibility.

> | Prop | Applied to | Reason  |

Reviewer 2: nit 

```suggestion
| Prop | Applied to | Reason |
```

| -- | -- | -- |

> |`screenReaderText`| `Banner` | Text announced by screen readers to indicate the type of banner. Defaults to `${variant} banner` if this prop is not passed in. |

Reviewer 3: This has been updated in one of the recent React PRs that got merged so that it not longer has a default. The current prop description is "Text announced by screen readers to indicate the type of banner. This prop should only be passed in when the banner conveys status/severity." Maybe:

```suggestion
Adds text that is accessible only to assistive technologies and is not visually rendered. This prop should only be passed in when the banner is meant to convey status.
```

Or something along those lines.

> |`screenReaderText`| `Banner` | Adds text that is accessible only to assistive technologies and is not visually rendered. This prop should only be passed in when the banner is meant to convey status. |

Reviewer 2: nit - spacing

```suggestion
| `screenReaderText` | `Banner` | Adds text that is accessible only to assistive technologies and is not visually rendered. This prop should only be passed in when the banner is meant to convey status. |
```

## HTML/CSS customization

The following HTML attributes and PatternFly classes can be used for more fine-tuned control over accessibility.

| Attribute or class | Applied to | Reason | 
|---|---|---|

> | `.pf-screen-reader` | `span` | Provides an accessible description of the type of banner. **Required if banner conveys status** |

Author: Sometimes "a" is appropriate and sometimes "the" is correct. For example, `A [as in any, use the indefinite article] banner should meet the following criteria:` versus `describe the contents of the [this particular one, use the definite article] banner`.
